python/cudf/cudf/core/column/string.py

```diff
@@ -847,7 +847,11 @@ def split(self, pat=None, n=-1, expand=True, **kwargs):
         NotImplementedError if anything other than the default value is set.
         """
         if expand is not True:
-            raise NotImplementedError("`expand` parameter is not supported")
+            raise NotImplementedError(
+                "`expand=False` setting is not supported yet"
+            )
+
+        warnings.warn("`expand` parameter defaults to True.")
```

Comment:
Maybe we should make `expand=None` the default and it coerces `None` --> `True` for now with a warning, but if the user explicitly specifies `expand=True` we shouldn't warn.